lxd/instance/drivers/driver_lxc.go

```diff
@@ -3636,6 +3636,9 @@ func (c *lxc) Rename(newName string) error {
 		}
 	}
 
+	// Set the new name in the struct.
+	c.name = newName
```

Comment:
@monstermunchkin one thing to consider though here is that before this change, c.name was set to the new name at a point where this function cannot fail. However where you have moved it now, its possible to fail and return with an error yet still leave the container struct with the new name set incorrectly.

There's probably a lot more that could be done in this function to allow a clean revert if an error occurs earlier, but at least not introducing any new potential inconsistencies is worth considering.

How about using the `revert` package and adding a revert step here that resets the name back to the original if this functions fails?

Reply:
That's a good point. I'll add some revert steps.